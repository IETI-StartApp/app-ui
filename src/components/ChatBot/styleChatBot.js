import styled from 'styled-components';

export const ButtonDesign = styled.button`
width: 100%;
height: 50px;
margin-top: 1px;
border-radius: 2px;
background: linear-gradient(90deg,
rgb(149, 7, 64) 0%,
rgb(127, 6, 54) 100%);
outline: none;
border: none;
align-content: center;
color: #fff;
font-size: 1rem;

`;
