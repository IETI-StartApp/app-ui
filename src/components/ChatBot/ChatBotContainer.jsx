import React from 'react'
import ChatBot from 'react-simple-chatbot';
import { Link } from 'react-router-dom'
import { ThemeProvider } from 'styled-components';
import { Grid } from '@material-ui/core';
import { ButtonDesign} from "./styleChatBot";
export const ChatBotContainer = () => {
  const theme = {
    
    background: '#f5f8fb',
    fontFamily: 'Helvetica Neue',
    headerBgColor: '#950740',
    headerFontColor: '#fff',
    headerFontSize: '15px',
    botBubbleColor: '#950740',
    botFontColor: '#fff',
    userBubbleColor: '#fff',
    userFontColor: '#4a4a4a',
  };

  const steps = [
    {
      id: '1',
      message: 'Bienvenido a StartApp, que quieres hacer hoy?',
      trigger: '2',
    },
    {
      id: '2',
      options: [
        { value: 1, label: 'Ver Proyectos', trigger: '3' },
        { value: 2, label: 'Ver mis proyectos', trigger: '4' },
        { value: 4, label: 'Agregar Proyecto', trigger: '5' },
       
      ],
    },
    {
      id: '4',
      component: (
        <Link to="/#features">
          <ButtonDesign className="button" >
            Ver mis proyectos!
          </ButtonDesign>
        </Link>
      ),
      end: true,
    },
    {
      id: '3',
      component: (
        <Link to="projects">
          <ButtonDesign className="button">
            Ver proyectos!
          </ButtonDesign>
        </Link>
      ),
      end: true,
    },
    {
      id: '5',
      component: (
        <Link to="register-project">
          <ButtonDesign className="button" >
            Agregar Proyecto!
          </ButtonDesign>
        </Link>
      ),
      end: true,
    },
  ]
  return (
    <div>
      <Grid container
                direction="row"
                justify="flex-end"
                alignItems="stretch"
                style={{ overflowY: "auto" }}>
                  <ThemeProvider theme={theme}>
        <ChatBot steps={steps} />;
            </ThemeProvider>
      </Grid>
      
    </div>
  )
}


