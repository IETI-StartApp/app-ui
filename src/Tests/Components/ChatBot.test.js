import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { ChatBotContainer } from '../../components/ChatBot/ChatBotContainer';

describe('Testing comment page', () => {
    const wrapper = shallow(<ChatBotContainer/>);
    
    test('should render comment page', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
})