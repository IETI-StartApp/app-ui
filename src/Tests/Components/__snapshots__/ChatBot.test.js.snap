// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing comment page should render comment page 1`] = `
<div>
  <WithStyles(ForwardRef(Grid))
    alignItems="stretch"
    container={true}
    direction="row"
    justify="flex-end"
    style={
      Object {
        "overflowY": "auto",
      }
    }
  >
    <Component
      theme={
        Object {
          "background": "#f5f8fb",
          "botBubbleColor": "#950740",
          "botFontColor": "#fff",
          "fontFamily": "Helvetica Neue",
          "headerBgColor": "#950740",
          "headerFontColor": "#fff",
          "headerFontSize": "15px",
          "userBubbleColor": "#fff",
          "userFontColor": "#4a4a4a",
        }
      }
    >
      <t
        avatarStyle={Object {}}
        botAvatar="data:image/svg+xml,%3csvg version='1' xmlns='http://www.w3.org/2000/svg' viewBox='0 0 512 512'%3e%3cpath d='M303 70a47 47 0 1 0-70 40v84h46v-84c14-8 24-23 24-40z' fill='%2393c7ef'/%3e%3cpath d='M256 23v171h23v-84a47 47 0 0 0-23-87z' fill='%235a8bb0'/%3e%3cpath fill='%2393c7ef' d='M0 240h248v124H0z'/%3e%3cpath fill='%235a8bb0' d='M264 240h248v124H264z'/%3e%3cpath fill='%2393c7ef' d='M186 365h140v124H186z'/%3e%3cpath fill='%235a8bb0' d='M256 365h70v124h-70z'/%3e%3cpath fill='%23cce9f9' d='M47 163h419v279H47z'/%3e%3cpath fill='%2393c7ef' d='M256 163h209v279H256z'/%3e%3cpath d='M194 272a31 31 0 0 1-62 0c0-18 14-32 31-32s31 14 31 32z' fill='%233c5d76'/%3e%3cpath d='M380 272a31 31 0 0 1-62 0c0-18 14-32 31-32s31 14 31 32z' fill='%231e2e3b'/%3e%3cpath d='M186 349a70 70 0 1 0 140 0H186z' fill='%233c5d76'/%3e%3cpath d='M256 349v70c39 0 70-31 70-70h-70z' fill='%231e2e3b'/%3e%3c/svg%3e"
        botDelay={1000}
        bubbleOptionStyle={Object {}}
        bubbleStyle={Object {}}
        cache={false}
        cacheName="rsc_cache"
        className=""
        contentStyle={Object {}}
        customDelay={1000}
        customStyle={Object {}}
        enableMobileAutoFocus={false}
        enableSmoothScroll={false}
        floating={false}
        floatingIcon={<ot />}
        floatingStyle={Object {}}
        footerStyle={Object {}}
        headerTitle="Chat"
        height="520px"
        hideBotAvatar={false}
        hideHeader={false}
        hideSubmitButton={false}
        hideUserAvatar={false}
        inputAttributes={Object {}}
        inputStyle={Object {}}
        placeholder="Type the message ..."
        recognitionEnable={false}
        recognitionLang="en"
        recognitionPlaceholder="Listening ..."
        speechSynthesis={
          Object {
            "enable": false,
            "lang": "en",
            "voice": null,
          }
        }
        steps={
          Array [
            Object {
              "id": "1",
              "message": "Bienvenido a StartApp, que quieres hacer hoy?",
              "trigger": "2",
            },
            Object {
              "id": "2",
              "options": Array [
                Object {
                  "label": "Ver Proyectos",
                  "trigger": "3",
                  "value": 1,
                },
                Object {
                  "label": "Ver mis proyectos",
                  "trigger": "4",
                  "value": 2,
                },
                Object {
                  "label": "Agregar Proyecto",
                  "trigger": "5",
                  "value": 4,
                },
              ],
            },
            Object {
              "component": <Link
                to="/#features"
              >
                <styled.button
                  className="button"
                >
                  Ver mis proyectos!
                </styled.button>
              </Link>,
              "end": true,
              "id": "4",
            },
            Object {
              "component": <Link
                to="projects"
              >
                <styled.button
                  className="button"
                >
                  Ver proyectos!
                </styled.button>
              </Link>,
              "end": true,
              "id": "3",
            },
            Object {
              "component": <Link
                to="register-project"
              >
                <styled.button
                  className="button"
                >
                  Agregar Proyecto!
                </styled.button>
              </Link>,
              "end": true,
              "id": "5",
            },
          ]
        }
        style={Object {}}
        submitButtonStyle={Object {}}
        userAvatar="data:image/svg+xml,%3csvg viewBox='-208.5 21 100 100' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink'%3e%3ccircle cx='-158.5' cy='71' fill='%23F5EEE5' r='50'/%3e%3cdefs%3e%3ccircle cx='-158.5' cy='71' id='a' r='50'/%3e%3c/defs%3e%3cclipPath id='b'%3e%3cuse overflow='visible' xlink:href='%23a'/%3e%3c/clipPath%3e%3cpath clip-path='url(%23b)' d='M-108.5 121v-14s-21.2-4.9-28-6.7c-2.5-.7-7-3.3-7-12V82h-30v6.3c0 8.7-4.5 11.3-7 12-6.8 1.9-28.1 7.3-28.1 6.7v14h100.1z' fill='%23E6C19C'/%3e%3cg clip-path='url(%23b)'%3e%3cdefs%3e%3cpath d='M-108.5 121v-14s-21.2-4.9-28-6.7c-2.5-.7-7-3.3-7-12V82h-30v6.3c0 8.7-4.5 11.3-7 12-6.8 1.9-28.1 7.3-28.1 6.7v14h100.1z' id='c'/%3e%3c/defs%3e%3cclipPath id='d'%3e%3cuse overflow='visible' xlink:href='%23c'/%3e%3c/clipPath%3e%3cpath clip-path='url(%23d)' d='M-158.5 100.1c12.7 0 23-18.6 23-34.4 0-16.2-10.3-24.7-23-24.7s-23 8.5-23 24.7c0 15.8 10.3 34.4 23 34.4z' fill='%23D4B08C'/%3e%3c/g%3e%3cpath d='M-158.5 96c12.7 0 23-16.3 23-31 0-15.1-10.3-23-23-23s-23 7.9-23 23c0 14.7 10.3 31 23 31z' fill='%23F2CEA5'/%3e%3c/svg%3e"
        userDelay={1000}
        width="350px"
      />
      ;
    </Component>
  </WithStyles(ForwardRef(Grid))>
</div>
`;
